python/lsst/summit/utils/plotting.py

```diff
@@ -322,3 +314,191 @@ def plot(
         plt.savefig(savePlotAs)
 
     return figure
+
+
+def _computeMtf(image: np.ndarray, midtonesBalance: float) -> np.ndarray:
+    """
+    Compute the midtones transfer function (MTF) for an image.
+
+    Parameters
+    ----------
+    image : `np.ndarray`
+        Input image normalized to [0, 1].
+    midtonesBalance : `float`
+        Balance parameter controlling midtones emphasis.
+
+    Returns
+    -------
+    image : `np.ndarray`
+        Image after applying the MTF, with NaNs replaced by fallback values.
+    """
+    M = np.full(image.shape, midtonesBalance)
+    maskHalf = image == 0.5
+    maskZero = image == 0
+    maskOne = image == 1
+    fallback = (maskHalf * 0.5) * (1 - maskZero) + maskOne
+    result = (M - 1) * image / ((2 * M - 1) * image - M)
+    nanMask = np.isnan(result)
```

Comment:
Perhaps use `~np.isfinite(result)` to catch inf values too.

Reply:
Good shout, thanks 🙏